src/controllers/redeploy.go

```diff
@@ -16,7 +16,8 @@ import (
 func redeploy(params *deployAction) {
 	state, _ := history.GetState(params.data["git_repo"].(string))
 
-	if state.Status == "stopped" {
+	if state.Status == "stopped" || state.Access == "" || state.Server == "" || state.Subdomain == "" {
+		log.Info("Repo is probably not yet deployed. Deploying now ..\n")
 		deploy(params)
 		return
```

Comment:
This wouldn't work since deploy would also require `access`, `server` and `subdomain` values. So if you see any of these values as empty, just log an error and return.
Also, remove the `state.Status == "stopped"` condition since that can happen organically.

Reply:
The `access`, `server` and `subdomain` fields are set by the deploy method using the `ActionInstance` object passed to it ([internalDeploy](https://github.com/devclub-iitd/DeployBot/blob/9bea7720a158ac205f69249394dddf65f9608bfa/src/controllers/deploy.go#L71)) which is in turn generated by the data obtained from slack request ([history.NewAction](https://github.com/devclub-iitd/DeployBot/blob/9bea7720a158ac205f69249394dddf65f9608bfa/src/controllers/deploy.go#L19)). In fact the `deploy` method is the first one to set up these state parameters for the service as evident by the [history.setState](https://github.com/devclub-iitd/DeployBot/blob/9bea7720a158ac205f69249394dddf65f9608bfa/src/controllers/deploy.go#L75) call. 

Also, the check for a `stopped` state was specifically added to avoid adding the `-x` flag to the deploy script which skips nginx regeneration which is undesirable since a stopped service doesn't have a nginx entry. In terms of abstractions, I think it makes more sense to "deploy" a stopped service before "re-deploying" it. So even if the methods for `redeploy` and `deploy` were to change in the future the logical flow would ensure that a stopped service gets deployed before being redeployed.

Reply:
If you look at this [code](https://github.com/devclub-iitd/DeployBot/blob/26c87aa936d980eef0e164fca54e6d2152beb93b/src/slack/config.go#L122-L144), you'll see that the redeploy request is not sent with any data for `access`, `server` and `subdomain` fields. The purpose of `redeploy` command is to rebuild and run a service by taking these values from its previous state. 

You're right that those fields are set by deploy method using the `ActionInstance` object, which takes its values from the slack request, but as I said above, in a `redeploy` request these values are not sent. So when you come across empty state values, there is no way to get the values for `access`, `server` and `subdomain` fields for deploying.

Reply:
Got it, Thanks!